substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1658,6 +1646,9 @@ pub mod pallet {
 					if let Some(ref mut nom) = maybe_nom {
 						if let Some(pos) = nom.targets.iter().position(|v| v == stash) {
 							nom.targets.swap_remove(pos);
+							// update nominations and trickle down to target list score.
+							Self::do_add_nominator(&nom_stash, nom.clone());
```

Comment:
Seems really weird to call `do_add_nominator` here. Probably rename it to `update_nominator`?

Reply:
I kinda agree, but it's been the conventions to call `do_add_*` to add/update nomiantor and validator  state. Also we have the comments here and in `fn do_add_nominator` which should make it clear.